### ISSUE TITLE
Github-actions - ansible-lint "InventoryModule" error

### ISSUE THREAD

**Issue by RamblingPSTech** [ansible/ansible-lint#4446]
### Summary

When using ansible-lint with github actions, ansible-lint fails with the following error when certain conditions are met:

`AttributeError: module 'ansible.plugins.inventory.host_list' has no attribute 'InventoryModule'`

Condition:

1. The inventory file is defined in `ansible.cfg`
2. One or more roles exist containing one or more yml files with or without content

### Issue Type

 - Bug Report

### OS / ENVIRONMENT

github actions using ansible/ansible-lint@main # (6fbb12e9d8e8464fe8dbc36f1829fa6a8d229c08)

The latest release v24.10.0 does not seem to be affected by this issue.

### STEPS TO REPRODUCE

1. Create basic inventory file `inventory.yml`

    ```yaml
    ---
    all:
      hosts:
        server1:
    ...

    ```

2. Create basic ansible confiuration file `ansible.cfg`

    ```ini
    [defaults]
    inventory = ./inventory.yml

    ```
3. Create basic playbook file `playbook.yml`

    ```yaml
    ---
    - name: Inventory ansible-lint with role
      hosts: server1
      gather_facts: false

      tasks:
        - name: Hello World
          ansible.builtin.debug:
            msg: "Hello World!"

    ```

4. Create basic role main.yml file `roles/common/tasks/main.yml`

    ```yaml
    ---
    # role/common/tasks/main.yml

    ```
5. Create basic github action workflow `.github/workflows/ansible-lint.yml`

    ```yaml
    # .github/workflows/ansible-lint.yml
    name: ansible-lint
    on:
      pull_request:
        branches: ["main", "stable", "release/v*"]
    jobs:
      build:
        name: Ansible Lint # Naming the build is important to use it as a status check
        runs-on: ubuntu-24.04
        steps:
          - uses: actions/checkout@v4
          - name: Run ansible-lint
            uses: ansible/ansible-lint@6fbb12e9d8e8464fe8dbc36f1829fa6a8d229c08

            # Also fails with "InventoryModule" error
            # uses: ansible/ansible-lint@main

            # optional (see below):
            with:
              args: ""
              setup_python: "true"
              working_directory: ""
              requirements_file: ""

    ```
6. Create basic README.md file `README.md`

    ```markdown
    # actions-test

    Testing ansible-lint actions

    ```

### Desired Behavior

`Passed: 0 failure(s), 0 warning(s) on 6 files. Last profile that met the validation criteria was 'production'.`

### Actual Behavior

github action should succeed or report any linting errors.

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.10/x64/bin/ansible-lint", line 8, in <module>
    sys.exit(_run_cli_entrypoint())
             ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansiblelint/__main__.py", line 400, in _run_cli_entrypoint
    sys.exit(main(sys.argv))
             ^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansiblelint/__main__.py", line 352, in main
    result = get_matches(rules, options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansiblelint/runner.py", line 751, in get_matches
    matches.extend(runner.run())
                   ^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansiblelint/runner.py", line 165, in run
    matches = self._run()
              ^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansiblelint/runner.py", line 265, in _run
    return_list = pool.map(worker, files, chunksize=1)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/multiprocessing/pool.py", line 367, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/multiprocessing/pool.py", line 774, in get
    raise self._value
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
                    ^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
           ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansiblelint/runner.py", line 247, in worker
    return self._get_ansible_syntax_check_matches(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansiblelint/runner.py", line 353, in _get_ansible_syntax_check_matches
    for inventory_file in self._get_inventory_files(app)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansiblelint/runner.py", line 502, in _get_inventory_files
    ).parse_source(
      ^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansible/inventory/manager.py", line 278, in parse_source
    for plugin in self._fetch_inventory_plugins():
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansible/inventory/manager.py", line 206, in _fetch_inventory_plugins
    plugin = inventory_loader.get(name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansible/plugins/loader.py", line 861, in get
    return self.get_with_context(name, *args, **kwargs).object
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/ansible/plugins/loader.py", line 912, in get_with_context
    obj = getattr(self._module_cache[path], self.class_name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'ansible.plugins.inventory.script' has no attribute 'InventoryModule'
Command failed: got '1', expected '0'
Error: Process completed with exit code 1.
```

**Comment by cavcrosby** [comment]
I'll give this a more thorough look over the next few days.

**Comment by cbugneac-nex** [comment]
I've also started getting these error after upgrade to `v24.10.0`, stack trace is similar to the above in GH Actions.

**Comment by mrolli** [comment]
+1

**Comment by cavcrosby** [comment]
This should now be fixed with one of my most recent PRs. I appreciate you reporting this.

**Comment by thegcat** [comment]
We also had this problem in a GitLab on premise CI and I can confirm this fixed the issues, thanks.

**Comment by theexiile1305** [comment]
+1

**Comment by RamblingPSTech** [comment]
> This should now be fixed with one of my most recent PRs. I appreciate you reporting this.

Thank you for the quick response. The `uses: ansible/ansible-lint@main` works again and does not show the "InventoryModule" error.
